src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/CoreCLR/WebRequestPSCmdlet.CoreClr.cs

```diff
@@ -374,91 +392,117 @@ protected override void ProcessRecord()
+                                    {
+                                        ErrorRecord er = new ErrorRecord(new InvalidOperationException(), "MaximumRedirectExceeded", ErrorCategory.InvalidOperation, request);
+                                        er.ErrorDetails = new ErrorDetails(WebCmdletStrings.MaximumRedirectionCountExceeded);
+                                        WriteError(er);
+                                    }
                                 }
                             }
-                            if (!String.IsNullOrEmpty(detailMsg))
+                            catch (HttpRequestException ex)
                             {
-                                er.ErrorDetails = new ErrorDetails(detailMsg);
+                                ErrorRecord er = new ErrorRecord(ex, "WebCmdletWebResponseException", ErrorCategory.InvalidOperation, request);
+                                if (ex.InnerException != null)
+                                {
+                                    er.ErrorDetails = new ErrorDetails(ex.InnerException.Message);
+                                }
+                                ThrowTerminatingError(er);
                             }
-                            ThrowTerminatingError(er);
-                        }
-
-                        ProcessResponse(response);
-                        UpdateSession(response);
 
-                        // If we hit our maximum redirection count, generate an error.
-                        // Errors with redirection counts of greater than 0 are handled automatically by .NET, but are
-                        // impossible to detect programmatically when we hit this limit. By handling this ourselves
-                        // (and still writing out the result), users can debug actual HTTP redirect problems.
-                        if (WebSession.MaximumRedirection == 0) // Indicate "HttpClientHandler.AllowAutoRedirect == false"
-                        {
-                            if (response.StatusCode == HttpStatusCode.Found ||
-                                response.StatusCode == HttpStatusCode.Moved ||
-                                response.StatusCode == HttpStatusCode.MovedPermanently)
+                            if (_followRelLink)
                             {
-                                ErrorRecord er = new ErrorRecord(new InvalidOperationException(), "MaximumRedirectExceeded", ErrorCategory.InvalidOperation, request);
-                                er.ErrorDetails = new ErrorDetails(WebCmdletStrings.MaximumRedirectionCountExceeded);
-                                WriteError(er);
+                                if (!_relationLink.ContainsKey("next"))
+                                {
+                                    return;
+                                }
+                                Uri = new Uri(_relationLink["next"]);
```

Comment:
I think it we be better to use a local variable instead of modifying the cmdlet parameter.